ui/shared/ShortcutTable.jsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+
+import treeherder from '../js/treeherder';
+
+class ShortcutTable extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { $injector } = this.props;
+
+    if ($injector) {
+      this.$rootScope = $injector.get('$rootScope');
+    }
+
+    this.setOnscreenShortcutsShowing = this.setOnscreenShortcutsShowing.bind(this);
+  }
+
+  setOnscreenShortcutsShowing() {
+    this.$rootScope.setOnscreenShortcutsShowing(false);
+  }
+
+  render() {
+    const { onscreenShortcutsShowing } = this.props;
+
+    return (
+      <div className="card">
+        <div className="card-header">
+          <h3>Keyboard shortcuts
+
+            {
+              onscreenShortcutsShowing &&
```

Comment:
Now that `index.html` no longer uses `stop-propagation-on-left-click` for the shortcuts table, the user can click anywhere to close it. As such, this close button is probably unnecessary now - so we could remove it? What do you think, @camd?

As an added bonus, removing it means this component no longer needs to use `$rootScope` or the constructor etc, and can be converted to a functional component.

Reply:
That sounds great, imo.  I only wonder if there's an a11y issue with not having a close button?  But ``esc`` also clears the dialog.  So I say let's remove the ``X``, as you recommend.  :)

Reply:
> I only wonder if there's an a11y issue with not having a close button?

Ah good point. Perhaps in addition to `Esc` (which already works), perhaps we should allow the user to close the panel by pressing `?` again? (Which currently doesn't work)

Reply:
It's very interesting to make shortcuts in the application to enrich the user experience, but does not appear that `?` key doesn't is intuitive to close something?

Reply:
I think the `X` button make it explain that the user can close the modal and the `esc` shorcut incremente with a natural way.

Reply:
The reason for making `?` close the dialogue, is that it's the same button people will have pressed to open it. ie a toggle. I think that is intuitive.